### PR TITLE
feat(analysis/optimization): adds ability to choose optimization, and provide optimization name when saving

### PR DESCRIPTION
**Pull request by dmitryevseev**
https://trello.com/c/6h6PPdlj/313-2-add-discounting-to-optimization

### CODE REVIEW

File: client/source/js/modules/analysis/optimization-ctrl.js [hunk 1]
```diff
@@ -605,8 +600,32 @@ define(['./module', 'angular', 'd3'], function (module, angular, d3) {
...
+        closeReason: 'overrideoptimization'
+      };
+
+      var doSave = function (name) {
+        $http.post('/api/analysis/optimization/save', { name: name })
+          .success(updateGraphs);
+      };
+
+      modalService.showPrompt(title, label,
+        function (newOptimizationName) {
```
Comment (sebastianconcept) [inline client/source/js/modules/analysis/optimization-ctrl.js]: Dmitry is using the modal service I've introduced a long ago.
I'm almost got a tear of emotion :D

:trollface:

File: client/source/js/modules/ui/modal/modal-service.js [hunk 2]
```diff
@@ -63,14 +63,18 @@ define([
...
             templateUrl: 'js/modules/ui/modal/modal-prompt.html',
             controller: ['$scope', function ($scope) {
               $scope.title = title;
               $scope.label = label;
-              $scope.acceptButton = 'Yes';
-              $scope.rejectButton = 'No';
+              $scope.acceptButton = options.acceptButton || 'Yes';
+              $scope.rejectButton = options.rejectButton || 'No';
+              $scope.description = options.description || false;
+              $scope.closeReason = options.closeReason || 'close';
```
Comment (sebastianconcept) [inline client/source/js/modules/ui/modal/modal-service.js]: So it can be customised but it fallback to those defaults if needed. Nice!

**Comment by nikgraf** [comment]
@Ashalynd can you have a look?

**Comment by Ashalynd** [comment]
cool, it seems to be working now:)

**Pull request merged**